source/Halibut/Util/CancelOnDisposeCancellationToken.cs

```diff
@@ -34,7 +34,7 @@ public sealed class CancelOnDisposeCancellationToken : IAsyncDisposable
         readonly CancellationTokenSource cancellationTokenSource;
         bool disposed;
 
-        ConcurrentBag<Task>? tasks = null;
+        readonly List<Task> tasks = new();
```

Comment:
Creating a list should be well fast enough since the ctor for `List()` appears to just use a static to initialise itself. 

I think ConcurrentBag is relatively expensive to init because of all the thread local stuff it wants to create.